### PR TITLE
Feat/ add app's streak counter

### PR DESCRIPTION
**Pull request by simon-the-shark**
Screenshot_1728146623

### CODE REVIEW

File: lib/features/app_streak/data/streak_start_repo.dart [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import "../../../utils/timestamp.dart";
+import "timstamp_repo_mixin.dart";
+
+part "streak_start_repo.g.dart";
+
+@riverpod
+class StreakStartRepository extends _$StreakStartRepository
+    with TimestampRepository {
+  @override
+  final storeKey = "__app_streak_counter__streak_start_stamp";
```
Comment (mikolaj-jalocha) [inline lib/features/app_streak/data/streak_start_repo.dart:14]: What about parametrizing it in the config file?

File: lib/features/app_streak/presentation/app_streak_counter.dart
Comment (mikolaj-jalocha) [inline lib/features/app_streak/presentation/app_streak_counter.dart:1]: Maybe an overkill, but I thought about adding simple loading shimmer here (it's everywhere else)

**Review by mikolaj-jalocha — APPROVED** [review]
LGTM! 

I quickly grasped what the code was about, so nice work. Everything nice & clean along with reusable mixin. I've left 2 comments.

File: lib/features/app_streak/presentation/app_streak_counter.dart
Comment (simon-the-shark) [reply lib/features/app_streak/presentation/app_streak_counter.dart:1]: it should basically be almost instantenious - no actualy loading should be seen imo in this case (shared preferences should be already loaded)

File: lib/features/app_streak/data/streak_start_repo.dart (hunk 1 above)
Comment (simon-the-shark) [reply lib/features/app_streak/data/streak_start_repo.dart:14]: sure

**Pull request merged**